### ISSUE TITLE
Parallel charge transfer efficiency issues

### ISSUE THREAD

**Issue by rongpu** [desihub/desispec#1956]
In #1946 we found several CCDs with parallel transfer issues. Suspecting that there are more such cases, I have done a somewhat systematic search for the parallel CTE feature by visually inspecting an arc exposure. The arc exposure that I inspected is EXPID=111642 from 20211201: `/global/cfs/cdirs/desi/spectro/redux/himalayas/preproc/20211201/00111642/`. Below are the (possible) parallel CTE cases that I found.

b3: x=3429 amp D
b7: x=2255 amp B; column x=2254 has low counts
b9: x=3286 amp D (visible at y=2070); I'm unsure if it's CTE
r0: x=1169 amp A
r2: x=1017 in amp A, x=2070 in amp B, x=1345 and x=1818 in amp C
r4: x=3919 amp D
r5: x=3876 amp D
r7: x=3743 and x=3923 amp D
r9: x=1813 amp C
z2: x=1786 amp A
z4: x=3383 amp B, x=2594 amp D
z5: x=1197 and 1199 amp A, x=2571 amp B

A few notes:
1. The coordinates are from ds9 so they are in 1-based indexing.
2. This search should have caught the most obvious cases, but might have missed some of the more subtle ones. Someone might be able to come up an automatic detection method that's better than visual inspection. Moreover, for B camera, because there are fewer arc lines, especially near the center of the image, if there is a bad charge transfer pixel near the center it might not be easily identified.
3. I haven't checked the pixel masks yet nor the masked fibers, so some of them might have already been masked.

**Comment by rongpu** [comment]
I have updated the pixmask files (in `desi_spectro_calib/trunk/ccd/`) and added masks for all the above CTE columns except for b9 (SVN revision 1068). All the pixmask files that cover the date 20210514 (the first day of the main survey) or after that are modified. (There are two CCDs, r4 and z5, that were replaced recently, and the pixmasks for the new CCDs are unchanged.) I have checked that all the CTE issues already existed on 20210514. 

I try to find the first pixel that has the charge transfer issue (usually in the LED flats rather than the arcs), and I only mask the subset of the rows up to that pixel. But sometimes it's hard to find that pixel (or it's clear from the arcs that almost all rows are affected), in which case I mask the whole column (of that amplifier). The notebooks that I used to apply and validate the masks can be found here: https://github.com/rongpu/desi-misc/tree/master/misc/desi_parallel_cte.

Note that we will need to reapply the parallel CTE masks (except for the new CCDs) whenever a new set of pixmask is generated.
